src/main/java/com/minecolonies/tileentities/ScarecrowTileEntity.java

```diff
@@ -0,0 +1,100 @@
+package com.minecolonies.tileentities;
+
+import com.minecolonies.inventory.InventoryField;
+import com.minecolonies.util.LanguageHandler;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+
+import java.util.Random;
+
+/**
+ * The scarecrow tile entity to store extra data.
+ */
+public class ScarecrowTileEntity extends TileEntity
+{
+    /**
+     * NBTTag to store the type.
+     */
+    private static final String TAG_TYPE = "type";
+    /**
+     * Random generator.
+     */
+    private final Random random = new Random();
+    /**
+     * The inventory connected with the scarecrow.
+     */
+    private InventoryField inventoryField;
+    /**
+     * The type of the scarecrow.
+     */
+    private ScareCrowType  type;
+
+    /**
+     * Empty public constructor.
+     */
+    public ScarecrowTileEntity()
+    {
+        super();
+        setInventoryField(new InventoryField(LanguageHandler.getString("com.minecolonies.gui.inventory.scarecrow"), true));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this call from a constructor to the overridable "setInventoryField" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1699)